### ISSUE TITLE
[kafka] Unhandled errors from RdKafka extension

### ISSUE THREAD

**Issue by Poltynik** [php-enqueue/enqueue-dev#986]
If there is any error from rdkafka extension like this one 
```
1573035921.906|FAIL|rdkafka#consumer-1| [thrd:kafka:9092/bootstrap]:kafka:9092/bootstrap: Failed to resolve 'kafka:9092':Name does not resolve
```
it can't be handled or reached.

https://github.com/php-enqueue/enqueue-dev/blob/master/pkg/rdkafka/RdKafkaConsumer.php#L165

Maybe some exception can be thrown here.

**Comment by Steveb-p** [comment]
@Poltynik thanks for creating this issue :)

From what I see the simplest solution would be to remove `RD_KAFKA_RESP_ERR__TIMED_OUT` from the `switch` cases list (thus resulting in `default`, which is to throw an exception).

I'll try to see why it was introduced there in the first place.

**Comment by Poltynik** [comment]
@Steveb-p, thanks.
Will wail for this update.

**Comment by nick-zh** [comment]
@Steveb-p heya, if you need to supress the stdout, this should be doable with:
```
$conf->set('log_level', '0');
```

**Comment by Steveb-p** [comment]
@Poltynik I just got bitten by the same issue you've reported :smile: My client was offline for month without me noticing. Dang... :imp: 

I'll fix tonight.

**Comment by nick-zh** [comment]
Fair warning though, `RD_KAFKA_RESP_ERR__TIMED_OUT` is not a true error, it just means, we haven't gotten any messages for the timeout we set (maybe there weren't any new ones).
By supressing the error, if the broker is unreachable, we will lose this information and most probably only will get the timeout. Enqueue might handle this differently and it might be fine, just wanted to point it out, that this might be a potential problem.

**Comment by Steveb-p** [comment]
This should be fixed by 41069e6e1d3b3b5f9d06c9aa20035d1c26775854.

I'm closing for now. Post a message if there are any issues with it.
